MovingBlocks/Terasology
# [v2.0.0] Convert alwaysRelevant to use entity scope instead

### Contains

This removes the `alwaysRelevant` entity field, and makes the relevant methods look at the `alwaysRelevant` field on the entity's scope instead.

`GLOBAL` and `SECTOR` scope are `alwaysRelevant`, but `CHUNK` scope is not. The default scope is `CHUNK`, so the default for `alwaysRelevant` is still `false.`.

This also updates some entity-related tests, and fixes some problems found by them (now all of the entity creation code goes through the `EntityBuilder` (previously almost all of it did, but methods like `createEntityWithId()` didn't), so that the scope can be set properly without putting it in multiple places).


### How to test

Load the [TutorialSectors](https://github.com/Terasology/TutorialSectors/tree/remove-alwaysrelevant) module on the `remove-alwaysrelevant` branch, and observe that the entity doesn't need to be set to `alwaysRelevant`, because it has already been marked as sector-scope (it will still simulate when you have unloaded all of its chunks).


### Outstanding before merging

Note: this depends on #3022

Some internal methods on the `PojoEntityManager` had to be exposed for use by the `EntityBuilder`, so it might be worth having a look at how the classes are structured (e.g. extracting the `EntityBuilder` to be an interface, and having a `PojoEntityBuilder` to make the Pojo-specific entities. Or putting the `EntityBuilder`'s `bulid()` method back into the `EntityManager`, now that it handles all of the different scenarios (scope, id, etc.) in one method.).

(Also, while on the topic of the `EntityBuilder`, it's probably worth looking at whether the `EntityBuilder` and `EntityStore` can be combined. They both do the same thing, they just differ in that the `EntityBuilder` requires an `EntityManager` when it is created, but the `EntityStore` involves the `EntityManager` when the entities are being built.)

There's also a possible issue that the `OnAddedComponent` is still sent when the entity is re-created from storage. With the previous implementation, this wasn't sent when re-created with the `createEntityWithId()` method, but if `ignoringEntityId` is true in the `EntitySerializer`, it would still send that event (but I don't know if that is ever actually used).
